packages/auth-nextjs/src/pages/client.ts

```diff
@@ -0,0 +1,113 @@
+import { type NextRouter } from "next/router";
+import {
+  type BuiltinProviderNames,
+  NextAuthHelpers,
+  type NextAuthOptions,
+} from "../shared.client";
+
+export { type NextAuthOptions, type BuiltinProviderNames };
+
+export default function createNextPagesClientAuth(options: NextAuthOptions) {
+  return new NextPagesClientAuth(options);
+}
+
+export class NextPagesClientAuth extends NextAuthHelpers {
+  async emailPasswordSignIn(
+    router: NextRouter,
```

Comment:
Is there a reason we cannot just use the imported `next/router` here directly from the library? Or pass it in the constructor?

Reply:
Turns out we can. I had always thought you had to use the router object from the `useRouter` hook, since the `router` object import directly didn't seem to work, but it seems it does work if you use the default import 🤷:  https://github.com/vercel/next.js/discussions/17046